cmp/options.go

```diff
@@ -363,24 +365,75 @@ func (cm comparer) String() string {
 //
 // In other cases, the cmpopts.IgnoreUnexported option can be used to ignore
 // all unexported fields on specified struct types.
-func AllowUnexported(types ...interface{}) Option {
+func AllowUnexported(templates ...interface{}) Option {
```

Comment:
Instead of overloading the meaning of the parameter, how about a new AllowUnexportedPackagePrefix option?

Reply:
We're already lacking type safety here, so meh. I slightly prefer having it overloaded so that we can keep the documentation in one place.

Reply:
The documentation is in one place, but it's more confusing--I'm not concerned about type safety, but having this function do quite different things depending on what types you pass to it is muddled. "This function allows unexported fields on a type, that function allows unexported fields in a package (prefix)" is much clearer.

Reply:
Done. I named it "AllowUnexportedWithinPackages"